utils/plot_utils.py

```diff
@@ -182,17 +182,23 @@ def generate_segmentation_colormap(all_annotations_data):
     To account for numerical inaccuracies, it is best to center the plot range
     around the class ids, by setting cmin=-0.5 and cmax=max_class_id+0.5.
     """
-    max_class_id = max(
-        [annotation_class["class_id"] for annotation_class in all_annotations_data]
-    )
+    # On export to pixelized masks, we squeeze the class ids to the range 0 to number of classes -1
+    # While these ids are not reflected in the store, i.e.
+    # (annotation_class["class_id"] for annotation_class in all_annotations_data)
+    # may have larger ids than num_classes -1 and gaps
+    # this is not the case for segmented results or masks
+    max_class_id = len(all_annotations_data) - 1
```

Comment:
Not sure if this is related to this change, but the segmented mask have missing classes when loading previously segmented results if I don't manually pre-setup the annotations classes to match the classes in the mask.
For example, if I have previously segmented a dataset with 2 classes. After reloading, there is only 1 default class. If I select a trained model and proceed to "view segmentation overlay", the segmented mask shows 1 class only instead of 2.

Reply:
This is unrelated to the change, but something we should indeed address.

Currently it is assumed that the number of classes listed in the interface does match with the classes of the segmentation. While this does enable the user to e.g. change colors of the overlay *and* annotations through simply interacting with the existing interface for classes, it can cause inconsistencies if the user e.g. deletes a class, or reloads the interface, annotates with less classes than before, but then still loads an older result.

For previously segmented results, we could choose to load class names, class colors, and annotations (along with model parameters) from the segmented result instead.

Reply:
Agreed. Moving this to an issue.